ec/ecresource/extensionresource/schema.go

```diff
@@ -18,64 +18,151 @@
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/resourcevalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/terraform-provider-ec/ec/internal/planmodifier"
+
+	"github.com/elastic/terraform-provider-ec/ec/internal"
 )
 
-func newSchema() map[string]*schema.Schema {
-	return map[string]*schema.Schema{
-		"name": {
-			Type:        schema.TypeString,
-			Description: "Required name of the ruleset",
-			Required:    true,
-		},
-		"description": {
-			Type:        schema.TypeString,
-			Description: "Description for extension",
-			Optional:    true,
-		},
-		"extension_type": {
-			Type:        schema.TypeString,
-			Description: "Extension type. bundle or plugin",
-			Required:    true,
-		},
-		"version": {
-			Type:        schema.TypeString,
-			Description: "Eleasticsearch version",
-			Required:    true,
-		},
-		"download_url": {
-			Type:        schema.TypeString,
-			Description: "download url",
-			Optional:    true,
-		},
+// Ensure provider defined types fully satisfy framework interfaces
+var _ resource.Resource = &Resource{}
+var _ resource.ResourceWithConfigure = &Resource{}
+var _ resource.ResourceWithImportState = &Resource{}
+var _ resource.ResourceWithConfigValidators = &Resource{}
 
-		// Uploading file via API
-		"file_path": {
-			Type:         schema.TypeString,
-			Description:  "file path",
-			Optional:     true,
-			RequiredWith: []string{"file_hash"},
-		},
-		"file_hash": {
-			Type:        schema.TypeString,
-			Description: "file hash",
-			Optional:    true,
-		},
+func (r *Resource) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Attributes: map[string]tfsdk.Attribute{
+			"name": {
+				Type:        types.StringType,
+				Description: "Required name of the ruleset",
+				Required:    true,
+			},
+			"description": {
+				Type:        types.StringType,
+				Description: "Description for extension",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					planmodifier.DefaultValue(types.String{Value: ""}),
+				}},
+			"extension_type": {
+				Type:        types.StringType,
+				Description: "Extension type. bundle or plugin",
+				Required:    true,
+			},
+			"version": {
+				Type:        types.StringType,
+				Description: "Elasticsearch version",
+				Required:    true,
+			},
+			"download_url": {
+				Type:        types.StringType,
+				Description: "download url",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					planmodifier.DefaultValue(types.String{Value: ""}),
+				},
```

Comment:
Why does this make sense as a default?

Reply:
@pascal-hofmann , can you please help with an answer?

Reply:
This is the default from the upstream API. Plan modifier prevents difference between planned state and result after apply.